app/main/controller/similarity_controller.py

```diff
@@ -36,53 +36,26 @@ def get_body_for_request(self):
     @api.doc('Delete a text in the similarity database')
     @api.expect(similarity_request, validate=True)
     def delete(self):
-        model_key = 'elasticsearch'
-        if 'model' in request.json:
-            model_key = request.json['model']
         es = Elasticsearch(app.config['ELASTICSEARCH_URL'])
-        body = self.get_body_for_request()
-        conditions = []
-        for body_key in body.keys():
-            if body_key == 'context':
-                for key in body[body_key]:
-                    matches = []
-                    for key in body[body_key]:
-                        matches.append({
-                            'match': { 'context.' + key: body[body_key][key] }
-                    })
-                    context = {
-                            'nested': {
-                            'score_mode': 'none',
-                            'path': 'context',
-                            'query': {
-                                'bool': {
-                                    'must': matches
-                                }
-                            }
-                        }
-                    }
-                    conditions.append(context)
-            else:
-                conditions.append({'match': {body_key: body[body_key]}})
-        query = {
-            'query': {
-                'bool': {
-                    'must': conditions
-                }
-            }
-        }
-        return es.delete_by_query(app.config['ELASTICSEARCH_SIMILARITY'], query)
+        return es.delete(index=app.config['ELASTICSEARCH_SIMILARITY'], id=request.json["doc_id"])
         
     @api.response(200, 'text successfully stored in the similarity database.')
     @api.doc('Store a text in the similarity database')
     @api.expect(similarity_request, validate=True)
     def post(self):
         body = self.get_body_for_request()
         es = Elasticsearch(app.config['ELASTICSEARCH_URL'])
-        result = es.index(
-            body=body,
-            index=app.config['ELASTICSEARCH_SIMILARITY']
-        )
+        if request.json.get("doc_id"):
+            result = es.update(
+                id=request.json["doc_id"],
+                body={"doc": body},
+                index=app.config['ELASTICSEARCH_SIMILARITY']
+            )
+        else:
+            result = es.index(
+                body=body,
+                index=app.config['ELASTICSEARCH_SIMILARITY']
+            )
```

Comment:
If I'm not mistaken, the logic of this block is:

```
IF doc_id is passed
  update document with that doc_id
ELSE
  create a new document with random id
```

While what we want is:

```
IF doc_id is passed
  IF document with doc_id exists
    update document with that doc_id
  ELSE
    create document with that doc_id
ELSE
  create document with random id
```

Right?

Reply:
yep! Good catch. Bad me, programming after my usual hours